File: clients/ruby/spec/features/connect_spec.rb
# frozen_string_literal: true

require_relative "support/context/acceptance"

RSpec.describe "ruby client api: connect" do
  include_context "acceptance"

  it "connects and returns a client" do
    expect(Proc.connect(authorization)).to be_instance_of(Proc::Client)
  end
end

File: clients/javascript/__tests__/features/connect.spec.js
require("../support/acceptance");

const Proc = require("../../index");

test("connects with an authorization", () => {
  expect(() => {
    Proc.connect(process.env.SECRET);
  }).not.toThrow();
});

test("connects with an authorization and host", () => {
  expect(() => {
    Proc.connect(process.env.SECRET, {host: "proc.local"});
  }).not.toThrow();
});

test("connects with an authorization, host, and scheme", () => {
  expect(() => {
    Proc.connect(process.env.SECRET, {host: "proc.local", scheme: "http"});
  }).not.toThrow();
});
